SDWebImage/SDImageTransformer.m

```diff
@@ -18,7 +18,23 @@
     if (!key || !transformerKey) {
         return nil;
     }
-    return [[key stringByAppendingString:SDImageTransformerKeySeparator] stringByAppendingString:transformerKey];
+    // Find the file extension
+    NSURL *keyURL = [NSURL URLWithString:key];
+    NSString *ext = keyURL ? keyURL.pathExtension : key.pathExtension;
+    if (ext.length > 0) {
+        // For non-file URL
+        if (keyURL && !keyURL.isFileURL) {
+            // keep anything except path (like URL query)
+            NSURLComponents *component = [NSURLComponents componentsWithURL:keyURL resolvingAgainstBaseURL:NO];
+            component.path = [[[component.path.stringByDeletingPathExtension stringByAppendingString:SDImageTransformerKeySeparator] stringByAppendingString:transformerKey] stringByAppendingPathExtension:ext];
```

Comment:
@dreampiggy Good for fix! Can we use `stringWithFormat` to replace these `stringByAppendingString`? seems long statement.

Reply:
OK...Actually these two have no difference... 😅

Reply:
The last one is not simple `stringByAppendingString`. The implementation their will check whether the path is end with `.`, whether the path is end of `/`. It's a complicated regular exception in the implementation files. If change, I can only change the first wo into `stringWithFormat:`